tests/core/full_node/test_mempool.py

```diff
@@ -1878,3 +1932,198 @@ def test_parse_unknown_condition(self):
 
             with pytest.raises(ValidationError):
                 cost, args = parse_condition_args(SExp.to([]), opcode, False)
+
+
+# the following tests generate generator programs and run them through get_name_puzzle_conditions()
+
+COST_PER_BYTE = 12000
+MAX_BLOCK_COST_CLVM = 11000000000
+
+
+def generator_condition_tester(
+    conditions: str, safe_mode: bool = False, quote: bool = True, max_cost: int = MAX_BLOCK_COST_CLVM
+) -> NPCResult:
+    prg = f"(q ((0x0101010101010101010101010101010101010101010101010101010101010101 {'(q ' if quote else ''} {conditions} {')' if quote else ''} 123 (() (q . ())))))"  # noqa
+    print(f"program: {prg}")
+    program = SerializedProgram.from_bytes(binutils.assemble(prg).as_bin())
+    generator = BlockGenerator(program, [])
+    print(f"len: {len(bytes(program))}")
+    npc_result: NPCResult = get_name_puzzle_conditions(
+        generator, max_cost, cost_per_byte=COST_PER_BYTE, safe_mode=safe_mode
+    )
+    return npc_result
+
+
+class TestGeneratorConditions:
+    def test_duplicate_height_time_conditions(self):
+        # ASSERT_SECONDS_RELATIVE
+        # ASSERT_SECONDS_ABSOLUTE
+        # ASSERT_HEIGHT_RELATIVE
+        # ASSERT_HEIGHT_ABSOLUTE
+        for cond in [80, 81, 82, 83]:
+            # even though the generator outputs multiple conditions, we only
```

Comment:
Hold up. If I say `ASSERT_SECONDS_ABSOLUTE 100` in block n, then I say `ASSERT_HEIGHT_ABSOLUTE n+1` in block n+1, can that constrain whether `ASSERT_SECONDS_ABSOLUTE 140` in block n+1 is valid?

Is there a defined relationship between ASSERT_SECONDS_ABSOLUTE and ASSERT_HEIGHT_ABSOLUTE?

Reply:
I don't think there's any absolute relationship that couldn't be changed by tweaking consensus constants in certain ways. Maybe you could prove some stuff for certain values of the consensus constants. But this test isn't really exercising the whole consensus stack like that.

Reply:
yeah, these conditions are validated independently. the relative conditions are relative to the time/height where the coin we're spending was created, whereas the absolute ones aren't dependent on the spent coin at all